r/R/dplyr.R

```diff
@@ -31,6 +31,18 @@ arrow_dplyr_query <- function(.data) {
     return(.data)
   }
 
+  # Evaluating expressions on a dataset with duplicated fieldnames will error
+  if (anyDuplicated(names(.data))) {
+    abort(c(
+      "Duplicated field names",
+      x = paste0(
+        "The following field names were found more than once in the data: ",
+        paste(names(.data)[duplicated(names(.data))], collapse = ", ")
+      ),
+      i =  "All field names must be unique"
+    ))
+  }
```

Comment:
Not that it matters much but we can call names() and duplicated() less (and use our paste helper):

```suggestion
  dupes <- duplicated(names(.data))
  if (any(dupes)) {
    abort(c(
      "Duplicated field names",
      x = paste0(
        "The following field names were found more than once in the data: ",
        oxford_paste(names(.data)[dupes])
      ),
      i =  "All field names must be unique"
    ))
  }
```

Also, do we need `i =  "All field names must be unique"`? Is that something that `abort()` suggests/insists on? It seems redundant.

Reply:
Fair comment, it does not, and you're right it's redundant - have removed it now.